android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/ContactsModule.java

```diff
@@ -42,6 +42,53 @@
 public class ContactsModule extends ReactContextBaseJavaModule {
   private static final String TAG = ContactsModule.class.getSimpleName();
 
+  private static final String[] PROJECTION = new String[]{
```

Comment:
Can you make sure we really need the projections? From my quick check: this also works when passing null to ContentResolver's query but maybe I'm missing something?

Reply:
check.

good catch, definitely an un-used variable - a remnant from some testing on generisizing the "add-on fields" functionality

Reply:
I also meant the other ones, are they required? Like I mentioned passing `null` seems to behave just okay but I might be wrong.

Reply:
![image](https://user-images.githubusercontent.com/4986192/36404975-828570de-15b3-11e8-99f7-d342ec0f5e9c.png)

tests start failing when i commented out all of the PROJECTION add statements. where did you pass null exactly?

Reply:
https://github.com/expo/expo/pull/1349/files#diff-86247190873d660d66a885d9af8272dcR200
right here

Reply:
thanks for clarifying. see below quote from [android docs](https://developer.android.com/reference/android/content/ContentResolver.html#query(android.net.Uri, java.lang.String[], android.os.Bundle, android.os.CancellationSignal)):

> projection | String: A list of which columns to return. Passing null will return all columns, which is inefficient.

Reply:
okay makes sense, I got a bit lost int the docs. Anway, i think the projection still contains phone numbers, emails and addresses even if they're not in `fields`.

Reply:
@aalices updated phone numbers, emails and addresses to only be projection if requested. surprisingly had essentially no impact on performance, could look into including them by default in future